lib/asar.js

```diff
@@ -200,11 +200,15 @@ module.exports.extractAll = function (archive, dest) {
       const linkTo = path.join(relativePath, path.basename(file.link))
       fs.symlinkSync(linkTo, destFilename)
     } else {
-      // it's a file, extract it
-      const content = disk.readFileSync(filesystem, filename, file)
-      fs.writeFileSync(destFilename, content)
-      if (file.executable) {
-        fs.chmodSync(destFilename, '755')
+      // it's a file, try to extract it
+      try {
+        const content = disk.readFileSync(filesystem, filename, file)
+        fs.writeFileSync(destFilename, content)
+        if (file.executable) {
+          fs.chmodSync(destFilename, '755')
+        } catch {
```

Comment:
You're missing a `}` here. The `catch` is currently on the `if` statement

Reply:
You're right, sorry about that. I actually tested the code this time but it looks like it's still failing the automated tests and I can't see why.